app/src/main/kotlin/com/hippo/ehviewer/ui/scene/SearchBarScene.kt

```diff
@@ -231,102 +241,112 @@ fun SearchBarScreen(
         }
     }
 
-    var searchbarOfs by searchBarOffsetY
-    Scaffold(
-        topBar = {
-            SearchBar(
-                modifier = Modifier.fillMaxWidth().layout { measurable, constraints ->
-                    val placeable = measurable.measure(constraints)
-                    layout(placeable.width, placeable.height) {
-                        placeable.placeRelative(0, searchbarOfs)
+    val searchbarOfs by rememberUpdatedState(searchBarOffsetY)
+    val scrollAwayModifier = if (!active) {
+        Modifier.layout { measurable, constraints ->
+            val placeable = measurable.measure(constraints)
+            layout(placeable.width, placeable.height) {
+                placeable.placeRelative(0, searchbarOfs)
+            }
+        }
+    } else {
+        Modifier
+    }
+    Box(modifier = Modifier.fillMaxSize()) {
+        Scaffold(
+            topBar = {
+                // Placeholder, fill immutable SearchBar padding
+                Spacer(modifier = Modifier.safeDrawingPadding().height(40.dp))
+            },
+            floatingActionButton = {
+                val hiddenState by animateFloatAsState(
+                    targetValue = if (showSearchFab) 1f else 0f,
+                    animationSpec = tween(FAB_ANIMATE_TIME),
+                    label = "hiddenState",
+                )
+                FloatingActionButton(
+                    onClick = ::onApplySearch,
+                    modifier = Modifier.rotate(lerp(90f, 0f, hiddenState)).scale(hiddenState),
+                ) {
+                    Icon(imageVector = Icons.Default.Search, contentDescription = null)
+                }
+            },
+            content = content,
+        )
+        PullToRefreshContainer(
+            state = refreshState,
+            modifier = Modifier.align(Alignment.TopCenter).safeDrawingPadding()
+                .padding(top = 48.dp) then scrollAwayModifier,
+        )
```

Comment:
Double indicators in GalleryListScene.
And they have different color.

Reply:
> Double indicators in GalleryListScene. And they have different color.

This indicator will never show, it's always hidden by Searchbar

Reply:
> > Double indicators in GalleryListScene. And they have different color.
> 
> This indicator will never show, it's always hidden by Searchbar

Yeah, but it is there.

Reply:
> > Double indicators in GalleryListScene. And they have different color.
> 
> This indicator will never show, it's always hidden by Searchbar if no nestedScrollConnection

Reply:
> > > Double indicators in GalleryListScene. And they have different color.
> > 
> > 
> > This indicator will never show, it's always hidden by Searchbar
> 
> Yeah, but it is there.

We will finally migrate all scene to compose
It will finally be dropped

Reply:
```
val surfaceColor =
            context.theme.resolveColor(com.google.android.material.R.attr.colorSurface)
        val overlayColor =
            context.getColor(com.google.android.material.R.color.m3_popupmenu_overlay_color)
        val backgroundColor = ColorUtils.compositeColors(overlayColor, surfaceColor)
        (mCircleView.background as ShapeDrawable).paint.color = backgroundColor
        setColorSchemeColors(context.theme.resolveColor(com.google.android.material.R.attr.colorAccent))
```
These color doesn't exist in MaterialTheme
We need to calculate them with compose style

Reply:
> > > > Double indicators in GalleryListScene. And they have different color.
> > > 
> > > 
> > > This indicator will never show, it's always hidden by Searchbar
> > 
> > 
> > Yeah, but it is there.
> 
> We will finally migrate all scene to compose It will finally be dropped

Well, I don't think we need pull-to-refresh in downloads?
IMO it's better to make it optional.

Reply:
> > > > > Double indicators in GalleryListScene. And they have different color.
> > > > 
> > > > 
> > > > This indicator will never show, it's always hidden by Searchbar
> > > 
> > > 
> > > Yeah, but it is there.
> > 
> > 
> > We will finally migrate all scene to compose It will finally be dropped
> 
> Well, I don't think we need pull-to-refresh in downloads? IMO it's better to make it optional.

Done